### PR TITLE
Fix: Register Blocks in Creative Tab and Items

### PR DESCRIPTION
**Pull request by Steellgold**

### CODE REVIEW

**Review by SquashOnGithub — APPROVED** [review]

**Review by SquashOnGithub — APPROVED** [review]

**Review by FRBlueScreen — APPROVED** [review]

**Review by FRBlueScreen — APPROVED** [review]

**Comment by Steellgold** [comment]
wtf

**Comment by Steellgold** [comment]
@TwistedAsylumMC Can you check up for the merge ?

**Review by Unickorn — CHANGES_REQUESTED** [review]
Please do not change the code style!

**Comment by Steellgold** [comment]
Here is @Unickorn

File: src/block/CustomiesBlockFactory.php [hunk 1]
```diff
@@ -42,8 +46,10 @@ final class CustomiesBlockFactory {
 	 * @phpstan-var array<string, Block>
 	 */
 	private array $customBlocks = [];
+
```
Comment (Unickorn) [inline src/block/CustomiesBlockFactory.php:49]: You still unnecessary formatting in multiple files, like this one.

File: src/item/ItemComponentsTrait.php [hunk 2]
```diff
@@ -65,10 +66,17 @@ public function getComponents(): CompoundTag {
...
 		$this->componentTag = CompoundTag::create()
 			->setTag("components", CompoundTag::create()
 				->setTag("item_properties", CompoundTag::create()
+					->setInt("creative_category", match ($cti->getCategory()) {
+						CreativeInventoryInfo::CATEGORY_CONSTRUCTION => 1,
+						CreativeInventoryInfo::CATEGORY_NATURE => 2,
+						CreativeInventoryInfo::CATEGORY_EQUIPMENT => 3,
+						default => 4
+					})
+					->setString("creative_group", $cti->getGroup() ?? CreativeInventoryInfo::GROUP_MISC_FOOD)
```
Comment (Unickorn) [inline src/item/ItemComponentsTrait.php:79]: I'm not sure if we should be having any default behavior here. If creative category is not being set then we can assume they do not want it to be in creative? Why make things nullable?

File: src/item/CustomiesItemFactory.php [hunk 3]
```diff
@@ -61,28 +66,31 @@ public function getItemTableEntries(): array {
...
-		if(($componentBased = $item instanceof ItemComponents)) {
+		if (($componentBased = $item instanceof ItemComponents)) {
 			$componentsTag = $item->getComponents();
 			$componentsTag->setInt("id", $item->getId());
 			$componentsTag->setString("name", $identifier);
 			$this->itemComponentEntries[$identifier] = new ItemComponentPacketEntry($identifier, new CacheableNbt($componentsTag));
 		}
+
 		$this->itemTableEntries[$identifier] = new ItemTypeEntry($identifier, $item->getId(), $componentBased);
+		CreativeInventory::getInstance()->add($item);
```
Comment (Unickorn) [inline src/item/CustomiesItemFactory.php:93]: I'm not entirely sure if registering things in the creative inventory by default is a good idea. Is this customies' responsibility? @TwistedAsylumMC please check

File: src/item/CustomiesItemFactory.php [hunk 4]
```diff
@@ -61,28 +66,31 @@ public function getItemTableEntries(): array {
 	/**
 	 * Registers the item to the item factory and assigns it an ID. It also updates the required mappings and stores the
 	 * item components if present.
+	 * CreativeInventoryInfo not completely implemented yet.
```
Comment (Unickorn) [inline src/item/CustomiesItemFactory.php:69]: What is the point of adding this comment? Unimplemented things will not be pushed either way, please add proper documentation instead.

**Review by Unickorn — CHANGES_REQUESTED** [review]
Apologies if I sound rude, but you still have tons of unnecessary format changes and I'm still not convinced this behavior is a responsibility of Customies. This is all achievable from the plugin end if a plugin is using Customies.

File: src/item/CustomiesItemFactory.php (hunk 4 above)
Comment (Steellgold) [reply src/item/CustomiesItemFactory.php:69]: This is a comment I forgot to remove this one

File: src/item/CustomiesItemFactory.php (hunk 3 above)
Comment (Steellgold) [reply src/item/CustomiesItemFactory.php:93]: In fact, if you don't put a CreativeInventoryInfo it won't appear in any of the creative inventory boxes 

I already tested everything and sent screenshots to @TwistedAsylumMC on Discord 😅

File: src/item/CustomiesItemFactory.php (hunk 4 above)
Comment (Steellgold) [reply src/item/CustomiesItemFactory.php:69]: This is a comment I forgot to remove this one

**Comment by Steellgold** [comment]
@Unickorn And without wanting to be mean, because translated into English (I am French) I do not know if it is in the same sense

But most of the modifications such as the addition of the Creative Tab Info class, it's a suggestion that @TwistedAsylumMC gave me in MP after I made my first version with an array

File: src/block/CustomiesBlockFactory.php (hunk 1 above)
Comment (Steellgold) [reply src/block/CustomiesBlockFactory.php:49]: Sorry my IDE did it automatically and I must not have seen it when sending the commit

**Comment by SquashOnGithub** [comment]
Having tested the plugin in its entirety, it works very well and this complementary functionality and that beneficial and indispensable !  @Steellgold @Unickorn @TwistedAsylumMC

File: src/item/ItemComponentsTrait.php (hunk 2 above)
Comment (Steellgold) [reply src/item/ItemComponentsTrait.php]: https://github.com/Steellgold/Customies/commit/20fac38cfbb6a956940599889d592e7afb7ed3dc

**Comment by Steellgold** [comment]
@Unickorn Here I made null by default if the user does not want to categorize these items, but I still found it useful to have them in the creative inventory, because otherwise many would ask where the items are 

https://github.com/Steellgold/Customies/commit/20fac38cfbb6a956940599889d592e7afb7ed3dc

**Comment by Steellgold** [comment]
@Unickorn @TwistedAsylumMC any news?
